geomesa-z3/src/main/scala/org/locationtech/geomesa/curve/NormalizedDimension.scala

```diff
@@ -17,7 +17,7 @@ trait NormalizedDimension {
     require(x >= min && x <= max, s"Value out of bounds ([$min $max]): $x")
     math.ceil((x - min) / (max - min) * precision).toInt
   }
-  def denormalize(x: Double): Double = (x / precision) * (max - min) + min
+  def denormalize(x: Double): Double = (x / (precision + 1)) * (max - min) + min
```

Comment:
I checked this, round-tripped, across [0,2^31) and it looks like the `precision +1` should be `precision + 0.5d`

I would recommend that we add unit tests to validate `n == normalize(denormalize(n))` the `0` and `precision`